AlarmClockSpeedTest.cpp

```diff
@@ -0,0 +1,88 @@
+/* 
+ * File:   AlarmClockSpeedTest.cpp
+ * Author: Amanda Carbonari
+ * Created: January 6, 2015 9:00am
+ */
+
+#include <ctime>
+#include <iostream>
+#include <AlarmClock.h> // To test other implementations, change this to #include<header file> or #inlcude<AlarmClock.h.old> for example
+#include <chrono>
+#include <thread>
+using namespace std;
+using namespace std::chrono;
+typedef std::chrono::microseconds microseconds;
+typedef std::chrono::milliseconds milliseconds;
+typedef std::chrono::seconds seconds;
+
+
+template<typename T> void WaitForAlarmClockToExpire(AlarmClock<T>& alerter) {
+   while(!alerter.Expired());
+}
+
+template <typename T> void testReset(unsigned int sleep_time, unsigned int reset_portion=2) {
+   // Start overall measurements
+   // cout << "Creating Alarm Clock" << endl;
+   high_resolution_clock::time_point start_overall = high_resolution_clock::now();
+   AlarmClock<T> alerter(sleep_time);
+   // cout << "Starting clock and resetting" << endl;
+
+   // Sleep before restart, this can be changed to see the overhead for
+   // different reset periods
+   std::this_thread::sleep_for(microseconds(sleep_time/reset_portion));
+
+   // Start the reset measurement
+   high_resolution_clock::time_point start = high_resolution_clock::now();
+
+   alerter.Reset();
+   
+   // Get the ending reset measurment
+   high_resolution_clock::time_point end = high_resolution_clock::now();
+
+   // Calculate the reset time
+   auto reset_time = duration_cast<microseconds>(end - start).count();
+   
+   // cout << "Waiting for the clock to expire" << endl;
+   // Make sure the clock expires correctly after reset
+   WaitForAlarmClockToExpire(alerter);
+
+   // Get the ending overall time
+   high_resolution_clock::time_point end_overall = high_resolution_clock::now();
```

Comment:
Please add a log output for the total time till expiration without a Reset in between. 
You have this from reset_time to end_overall